python/lsst/pipe/base/pipelineIR.py

```diff
@@ -791,18 +793,52 @@ def from_string(cls, pipeline_string: str):
         return cls(loaded_yaml)
 
     @classmethod
-    def from_file(cls, filename: str):
+    def from_file(cls, filename: str) -> PipelineIR:
         """Create a `PipelineIR` object from the document specified by the
         input path.
 
         Parameters
         ----------
         filename : `str`
             Location of document to use in creating a `PipelineIR` object.
+
+        Returns
+        -------
+        pipelineIR : `PipelineIR`
+            The loaded pipeline
+
+        Note
+        ----
+        This method is deprecated, please use from_uri
         """
-        with open(filename, 'r') as f:
-            loaded_yaml = yaml.load(f, Loader=PipelineYamlLoader)
-        return cls(loaded_yaml)
+        return cls.from_uri(filename)
+
+    @classmethod
+    def from_uri(cls, uri: str) -> PipelineIR:
+        """Create a `PipelineIR` object from the document specified by the
+        input uri.
+
+        Parameters
+        ----------
+        uri: `str`
```

Comment:
As with `Pipeline.fromUri`, ideally this would probably take `ButlerURI` instances as well as `str` (and probably already actually does).